openstack_controller/datadog_checks/openstack_controller/openstack_controller.py

```diff
@@ -205,24 +205,37 @@ def get_loads_for_single_hypervisor(self, hyp_id):
         uptime = self.get_os_hypervisor_uptime(hyp_id)
         return self._parse_uptime_string(uptime)
 
-    def collect_hypervisors_metrics(self, custom_tags=None,
+    def collect_hypervisors_metrics(self, servers,
+                                    custom_tags=None,
                                     use_shortname=False,
                                     collect_hypervisor_metrics=True,
                                     collect_hypervisor_load=False):
         """
         Submits stats for all hypervisors registered to this control plane
         Raises specific exceptions based on response code
         """
+        # Create a dictionary with hypervisor hostname as key and the list of project names as value
+        hyp_project_names = {}
+        for _, server in iteritems(servers):
+            if not server['hypervisor_hostname']:
```

Comment:
Is there a chance that key would not exist? If so:
```suggestion
            if not server.get('hypervisor_hostname'):
```